Oracle.md

```diff
@@ -0,0 +1,135 @@
+
+#### Example: BTC/USD rate
+
+```
+base: 10
+exponent: 5
+mantissa: 0.9
+```
+
+### Serialization and signature
+
+Event templates should be serialized using [TLV format](https://github.com/lightningnetwork/lightning-rfc/blob/master/01-messaging.md#type-length-value-format) (to be defined).
+The oracle should provide a signature over the serialized template to the client as an attestation that he committed to release a signature over the event outcome.
+
+## Unique Resource Identifiers
+
+To identify, organize or filter events, it is useful that they are assigned unique resource identifier.
+
+### Categorization
+
+Each segment of the URI path, with the exception of the last one, should represent a (sub)-categorization of the event that is being identified.
+Upon being requested a URI corresponding to a category, the resource provider should return a list of resources included in the category that should correspond to either a sub-category or of an event together with their respective URI (HATEOAS).
+The interpretation of the category semantics is not specified and left to the interpretation of the client (or user).
+
+Note that in some cases the list of event might be to large to be enumerated.
+This is for example the case for financial events that might be released periodically over a long period of time.
+In that case, the oracle should return a way for the client for constructing the possible url.
+
+In the case where the events occur periodically, the oracle can return a date range represented as follow:
+
+```
+- startDate: the date of the first event to have occurred within the category (could be past or future) in ISO 8601 format
+- frequency: a period of time representing the frequency at which event in the category occur (ISO 8601 format)
+- range: how far in the future can the oracle provide a template for an event relative to the request time (period in ISO 8601 format).
+```
+
+For cases where the set of events are represented using a number, a range (start, end, step) suffices.
+
+#### Example: Oracle web server
+
+A client is given the hostname of an oracle web server `https://crystalball.dlc`.
+The client accesses the root resource `/` and is returned a list of event categories served by the oracle with their respective URIs:
+
+- Sport: `/sport`
+- Finance: `/finance`
+- Politics: `/politics`
+
+The client can then request these URIs to access events or sub-categories.
+
+If the oracle is supporting BTC/USD, a request over `/finance/cryptocurrencies/BTC_USD/biweekly` could for example return:
+
+```
+- startDate: "2020-08-07T22:00:00Z"
+- frequency: "P2W"
+- range: "P4W"
+```
+
+This means that the oracle will sign the BTC/USD rate every two weeks (on Fridays) and that a client can query up to 4 weeks in advance.
+
+### Events
+
+The last segment of a URI must identify a single event.
+Upon requesting an event URI, a client must be served with:
+
+- The event template for the event together with a signature over it
```

Comment:
I assume this includes R values as well?

Reply:
Yes!